testrunner.py

```diff
@@ -17,12 +18,40 @@ def test_status(self, expected_status, actual_status):
         In a separate function to make debugging easy with py.test
         """
         assert expected_status == actual_status
-        
-    def run_stage(self, stage):
+    
+    def test_log(lines, log_contains):
+        """
+        Checks if a series of log lines contains a regex specified in the
+        output stage. It will flag true on teh first log_contains regex match
+        and then assert on the flag at the end of the function
+        """
+        found = False
+        for line in lines:
+           if log_contains.search(line):
+            found = True
+            break
+        assert(found)
```

Comment:
You don't need to `assert` on `found`, if you break you know found was `True` in the first place.